relevanceai/vector_tools/plot_txt_gist.py

```diff
@@ -0,0 +1,202 @@
+
+        # get cluster data
+        cluster_data = self.get_cluster_datafield(docs = docs, vector_field = vector_field, text_field = text_field,
+            alias = alias, lower = lower, remove_digit= remove_digit, remove_punct = remove_punct)
+        clusters_top_n_words = self.get_cluster_word_freq(cluster_data = cluster_data, remove_stop_words = remove_stop_words,
+            additional_stop_words = additional_stop_words, cluster_representative_cnt = cluster_representative_cnt)
+        for c in clusters_top_n_words:
+            print(c, clusters_top_n_words[c])
+
+        # dimensionality reduction
+        # 1) Fit the model and reduce vector size
+        vectors = self.client.get_field_across_documents(vector_field, docs)
+        dr_docs = self.dim_reduction(vector_data = vectors, embedding_dims = self.embedding_dims, k = self.dim_red_k, 
+            n_epochs_without_progress = self.n_epochs_without_progress)
+        for i, dr in enumerate(dr_docs):
+              docs[i][vector_field +'_dr_vector_'] = dr.tolist()
+        # 2) reduce center vector size
+        vectors = [c['centroid_vector_'] for c in centers]
+        dr_centers = self.dim_reduction(vector_data = vectors, embedding_dims = self.embedding_dims, k = self.dim_red_k, 
+            n_epochs_without_progress = self.n_epochs_without_progress, fit = False)
+        for i, c in enumerate(centers):
+            c['centroid_dr_vector_'] = dr_centers[i]
+        
+        # plot
+        self.plot_clusters(docs, dr_docs, centers, cluster_data, vector_field, alias, cluster_representative_cnt)
+
+    def get_documents(self, vector_field:str, text_field:str, max_doc_num: int = None):
+        print(" * Loading documents")
+        filters=[{'field' : vector_field, 'filter_type' : 'exists', "condition":"==", "condition_value":" "}]
+        fields = [vector_field, text_field]
+        if max_doc_num:
+            page_size = 200 if max_doc_num > 200 else max_doc_num
+            batch_doc = self.batch_load_docs(fields = fields, filters = filters, page_size = page_size)
+            docs= batch_doc['documents']
+            cursor = batch_doc['cursor']
+            while batch_doc['documents'] != [] and len(docs) > max_doc_num - page_size:
+                batch_doc = self.batch_load_docs(fields = fields, filters = filters, page_size = page_size, cursor = cursor)
+                docs.extend(batch_doc['documents'])
+        else:
+            docs = self.client.get_all_documents(dataset_id = self.dataset_id, filters = filters, select_fields = fields)
+        return docs
+
+    def batch_load_docs(self, fields:List[str], filters:List[dict] = [], page_size:int = 200, cursor:str = None):
+        return self.client.datasets.documents.get_where(dataset_id=self.dataset_id, select_fields=fields, 
+                page_size= page_size, filters=filters, cursor = cursor)
+
+    def kmeans_clustering(self, docs, vector_field = str, k : int = 10, alias: str = "kmeans"):
```

Comment:
You can just use the internal one liner now

Reply:
I decided to run it step-by-step to also have clustered_docs (i.e. not just centers).